dist/src/main/java/io/camunda/application/configuration/GatewayBasedConfiguration.java

```diff
@@ -14,23 +14,90 @@
 import io.atomix.cluster.messaging.MessagingConfig;
 import io.atomix.cluster.protocol.SwimMembershipProtocolConfig;
 import io.atomix.utils.net.Address;
+import io.camunda.application.configuration.GatewayBasedConfiguration.GatewayBasedProperties;
+import io.camunda.commons.actor.ActorSchedulerConfiguration.SchedulerConfiguration;
+import io.camunda.commons.broker.client.BrokerClientConfiguration.BrokerClientTimeoutConfiguration;
+import io.camunda.commons.job.JobHandlerConfiguration.ActivateJobHandlerConfiguration;
 import io.camunda.zeebe.gateway.impl.configuration.ClusterCfg;
 import io.camunda.zeebe.gateway.impl.configuration.GatewayCfg;
 import io.camunda.zeebe.gateway.impl.configuration.MembershipCfg;
+import io.camunda.zeebe.gateway.impl.configuration.MultiTenancyCfg;
+import io.camunda.zeebe.gateway.rest.ConditionalOnRestGatewayEnabled.RestGatewayDisabled;
+import java.time.Duration;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.context.LifecycleProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
 
 @Configuration(proxyBeanMethods = false)
-public final class GatewayClusterConfiguration {
+@EnableConfigurationProperties(GatewayBasedProperties.class)
+@Profile("!broker")
+public final class GatewayBasedConfiguration {
```

Comment:
❓ How does this work with the embedded Gateway? I guess this is only for standalone gateway necessary?

Reply:
This does not affect the embedded gateway. The embedded gateway is started only with profile `broker` (and when `zeebe.broker.gateway.enable` is set to `true`). Since the `broker` profile is used for the embedded gateway, the application will use the `BrokerBasedConfiguration` and configure the embedded gateway accordingly (as done previously).

The condition `!broker` assumes that if the broker is not started with the application, then the node acts "only" as a "gateway," exposing the Camunda API, aka v2 endpoints (in addition to potentially serving a respective UI).

To sum up, the application configuration happens either by
1. `BrokerBasedConfiguration`: using the `zeebe.broker.*` configurations, or
2. `GatweayBasedConfiguration`: using the `zeebe.gateway.*` configurations.

Based on that, commons modules (i.e., the actor scheduler, Atomix cluster, etc.) are started so that the Camunda API can be exposed on any node.

Reply:
Thanks